crates/router/src/core/subscription.rs

```diff
@@ -63,3 +77,556 @@ pub async fn create_subscription(
+    )?;
+    let handler = SubscriptionHandler::new(state, merchant_context, request, profile_id);
+
+    let mut subscription_entry = handler
+        .find_subscription(subscription_id.get_string_repr().to_string())
+        .await?;
+
+    let billing_handler = subscription_entry.get_billing_handler().await?;
+    let invoice_handler = subscription_entry.get_invoice_handler().await?;
+
+    let _customer_create_response = billing_handler.create_customer(&handler.state).await?;
+
+    let subscription_create_response = billing_handler.create_subscription(&handler.state).await?;
+
+    // let payment_response = invoice_handler.create_cit_payment().await?;
+
+    let invoice_entry = invoice_handler
+        .create_invoice_entry(
+            &handler.state,
+            subscription_entry.profile.get_billing_processor_id()?,
+            None,
+            billing_handler.request.amount,
+            billing_handler.request.currency.to_string(),
+            common_enums::connector_enums::InvoiceStatus::InvoiceCreated,
+            billing_handler.connector_name.clone(),
+            None,
+        )
+        .await?;
+
+    // invoice_entry
+    //     .create_invoice_record_back_job(&payment_response)
+    //     .await?;
+
+    subscription_entry
+        .update_subscription_status(
+            SubscriptionStatus::from(subscription_create_response.status).to_string(),
+        )
+        .await?;
+
+    let response = subscription_entry
+        .generate_response(&invoice_entry.invoice, subscription_create_response.status)?;
+
+    Ok(ApplicationResponse::Json(response))
+}
+
+pub struct SubscriptionHandler {
+    state: SessionState,
+    merchant_context: MerchantContext,
+    request: subscription_types::ConfirmSubscriptionRequest,
+    profile_id: common_utils::id_type::ProfileId,
+}
+
+impl SubscriptionHandler {
+    pub fn new(
+        state: SessionState,
+        merchant_context: MerchantContext,
+        request: subscription_types::ConfirmSubscriptionRequest,
+        profile_id: common_utils::id_type::ProfileId,
+    ) -> Self {
+        Self {
+            state,
+            merchant_context,
+            request,
+            profile_id,
+        }
+    }
+    pub async fn find_subscription(
+        &self,
+        subscription_id: String,
+    ) -> errors::RouterResult<SubscriptionWithHandler<'_>> {
+        let subscription = self
+            .state
+            .store
+            .find_by_merchant_id_subscription_id(
+                self.merchant_context.get_merchant_account().get_id(),
+                subscription_id.clone(),
+            )
+            .await
+            .change_context(errors::ApiErrorResponse::GenericNotFoundError {
+                message: format!("subscription not found for id: {subscription_id}"),
+            })?;
+
+        let key_manager_state = &(&self.state).into();
+        let merchant_key_store = self.merchant_context.get_merchant_key_store();
+
+        let profile = self
```

Comment:
Move profile read outside find_subscription, profile read will be at confirm_subscription or auth layer itself can return profile, verify AuthenticationData for the same